crates/rpc-types-eth/src/block.rs

```diff
@@ -247,13 +249,14 @@ impl HeaderResponse for Header {
 }
 
 /// Error that can occur when converting other types to blocks
-#[derive(Clone, Copy, Debug, thiserror::Error)]
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
```

Comment:
the problem with this is that this now removes the display impl for no-std

perhaps we should use derive_more display and manually impl error for all errors @DaniPopes ?